webapp/channels/src/sass/layout/_headers.scss

```diff
@@ -301,6 +315,7 @@
 
     .channel-header__title {
         display: flex;
+        height: 24px;
 
```

Comment:
gap: 4px;

Recommend adding 4px space between the Channel title menu item and the icon button. Currently looks odd in the hover and selected state. 0/5 on how to do it, this is just one way, can be done via margins also. 
<img width="341" alt="image" src="https://github.com/mattermost/mattermost/assets/14369607/0f238f49-1df7-41d9-8929-accce387bbbe">

Reply:
Thanks @abhijit-singh I've updated this in the latest commit